src/molinillo_solver.cr

```diff
@@ -61,6 +62,8 @@ module Shards
               next unless dep.matches?(version)
             end
 
+            next if dep.resolver != lock.resolver
```

Comment:
This would resolve to the latest version, for example in the case of a url change. Think about moving from GitLab to GitHub, or renaming the repository. Should we try to keep the locked version and force an `update` in case it's not available?

Reply:
I see, you mean that 

1. if the shard.yml has no version restriction, 
2. there is a lock pointing to the non latest version
3. the shard.yml is changed to point to another fork (and still has no version restriction)

The result is that the lock is totally discarded and the dependency will bump.

I see a change of source as a change of dependency entirerly, so I wouldn't try to keep the lock.
I also have more in mind changing the source form git to path for local overrides, there the lock is also expected to be discarded.

Reply:
But that's a problem, because there is no way to avoid that behaviour. If the locked version is honoured, there is always the option to run an `update` to unlock it.

For example, if you're testing something and want to use a local version of the dependency momentarily, you don't want the whole subtree of dependencies to upgrade.

I think the solution would be as simple as using a new dependency as the lock node, built using the resolver coming from the spec dependency, combined with the locked version, and probably raise a warning or something to notify the change being made.

Reply:
If there is a change in the shard.yml that makes the lock not a candidate, the lock for that dependency is ignored.
I aimed for the same behaviour if the source change. The version might have differed completely between forks.

But maybe we can keep the locks of the subtree.... that would be more conservative.

Reply:
On other hand, the forks can use different dependencies. Keeping locked only the ones that will still be required, to allow the others to go away, will require resolving which version in the forked source will be used.

I'm more inclined to keep the current behavior in the PR. Unlocking all dependencies and treating the change of source as a complete change of the indicated dependency.